File: backend/src/main/java/com/backend/domain/repository/dto/response/github/RepoResponse.java
package com.backend.domain.repository.dto.response.github;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

public record RepoResponse(
        String name,
        @JsonProperty("full_name") String fullName,
        @JsonProperty("private") boolean _private,
        String description,
        @JsonProperty("html_url") String htmlUrl,
        String language,
        @JsonProperty("default_branch") String defaultBranch,
        @JsonProperty("created_at") OffsetDateTime createdAt,
        Integer size
) {
}

File: backend/src/main/java/com/backend/domain/repository/dto/response/github/RepoResponse.kt
package com.backend.domain.repository.dto.response.github

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime

data class RepoResponse(
    val name: String?,
    @JsonProperty("full_name")
    val fullName: String?,
    val isPrivate: Boolean,
    val description: String?,

    @JsonProperty("html_url")
    val htmlUrl: String?,
    val language: String?,

    @JsonProperty("default_branch")
    val defaultBranch: String?,

    @JsonProperty("created_at")
    val createdAt: OffsetDateTime?,
    val size: Int?
)
